pkg/reconciler/kubernetes/tektonchain/tektonchain.go

```diff
@@ -509,3 +513,41 @@ func (r *Reconciler) markUpgrade(ctx context.Context, tc *v1alpha1.TektonChain)
 	}
 	return v1alpha1.RECONCILE_AGAIN_ERR
 }
+
+func AddControllerEnv(controllerEnvs []v1alpha1.ControllerEnv) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "Deployment" {
+			return nil
+		}
+
+		d := &appsv1.Deployment{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, d)
+		if err != nil {
+			return err
+		}
+
+		if controllerEnvs != nil {
+			for _, v := range controllerEnvs {
+				env := corev1.EnvVar{
+					Name:  v.Name,
+					Value: v.Value,
+				}
+
+				for i, c := range d.Spec.Template.Spec.Containers {
+					if c.Name == "tekton-chains-controller" {
+						d.Spec.Template.Spec.Containers[i].Env = append(d.Spec.Template.Spec.Containers[i].Env, env)
```

Comment:
If a existing env found with the same name, replace that with new env.
otherwise append the new env.

Reply:
I guess deployment should give an error if key with same name are added 🤔

Reply:
but we dont want to error right. we will error in validation if two variable are defined with same key.

here we want to supply the one which is there on CR and not fail the installation

Reply:
also this will make the installation error if value is updated in CR

Reply:
Yeah got it 👍🏻